process/block/sovereign/outgoingOperations_test.go

```diff
@@ -25,115 +25,376 @@ func createEvents() []SubscribedEvent {
+				},
+			},
+		}
+
+		args := ArgsOutgoingOperations{
+			SubscribedEvents: events,
+			DataCodec:        &mock.DataCodecMock{},
+			TopicsChecker:    &mock.TopicsCheckerMock{},
+		}
+		creator, err := NewOutgoingOperationsFormatter(args)
+		require.Nil(t, creator)
+		require.ErrorContains(t, err, errNoSubscribedAddresses.Error())
+	})
+}
+
+func TestOutgoingOperations_OutgoingEvents(t *testing.T) {
+	t.Parallel()
+
+	logs := []*data.LogData{
+		{
+			LogHandler: &transactionData.Log{
+				Address: nil,
+				Events: []*transactionData.Event{
+					{
+						Address:    []byte("addr1"),
+						Identifier: []byte("deposit"),
+						Topics:     [][]byte{[]byte("topic1"), []byte("topic1"), []byte("topic1"), []byte("topic1"), []byte("topic1")},
+						Data:       []byte("data"),
+					},
+				},
+			},
+			TxHash: "",
+		},
+	}
+
+	t.Run("nil logs", func(t *testing.T) {
+		t.Parallel()
+
+		outgoingOps := createValidArgs()
+
+		outgoingTxData, err := outgoingOps.CreateOutgoingTxsData(nil)
+		require.NoError(t, err)
+		require.Equal(t, 0, len(outgoingTxData))
+	})
+	t.Run("deserialize token error", func(t *testing.T) {
+		t.Parallel()
+
+		outgoingOps := createValidArgs()
+
+		outgoingOps.dataCodec = &mock.DataCodecMock{
+			DeserializeTokenDataCalled: func(_ []byte) (*sovereign.EsdtTokenData, error) {
+				return nil, fmt.Errorf("deserialize token data error")
+			},
+		}
+
+		outgoingTxData, err := outgoingOps.CreateOutgoingTxsData(logs)
+		require.Nil(t, outgoingTxData)
+		require.Equal(t, "deserialize token data error", err.Error())
+	})
+	t.Run("deserialize event error", func(t *testing.T) {
+		t.Parallel()
+
+		outgoingOps := createValidArgs()
+
+		outgoingOps.dataCodec = &mock.DataCodecMock{
+			DeserializeEventDataCalled: func(data []byte) (*sovereign.EventData, error) {
+				return nil, fmt.Errorf("deserialize event data error")
+			},
+			DeserializeTokenDataCalled: func(_ []byte) (*sovereign.EsdtTokenData, error) {
+				return &sovereign.EsdtTokenData{}, nil
+			},
+		}
+
+		outgoingTxData, err := outgoingOps.CreateOutgoingTxsData(logs)
+		require.Nil(t, outgoingTxData)
+		require.Equal(t, "deserialize event data error", err.Error())
+	})
+	t.Run("serialize operation error", func(t *testing.T) {
+		t.Parallel()
+
+		outgoingOps := createValidArgs()
+
+		outgoingOps.dataCodec = &mock.DataCodecMock{
+			DeserializeEventDataCalled: func(data []byte) (*sovereign.EventData, error) {
+				return &sovereign.EventData{}, nil
+			},
+			DeserializeTokenDataCalled: func(_ []byte) (*sovereign.EsdtTokenData, error) {
+				return &sovereign.EsdtTokenData{}, nil
+			},
+			SerializeOperationCalled: func(operation sovereign.Operation) ([]byte, error) {
+				return nil, fmt.Errorf("serialize operation error")
+			},
+		}
+
+		outgoingTxData, err := outgoingOps.CreateOutgoingTxsData(logs)
+		require.Nil(t, outgoingTxData)
+		require.Equal(t, "serialize operation error", err.Error())
+	})
+}
+
+func TestOutgoingOperations_CheckValidity(t *testing.T) {
```

Comment:
Either rename this func or move it inside one of the error cases above, since you are testing `CreateOutgoingTxsData`